test/testlib/nuodb_admin_utilities.go

```diff
@@ -223,3 +224,53 @@ func ApplyNuoDBLicense(t *testing.T, namespace string, adminPod string) {
 		require.True(t, strings.Contains(output, "true"), "NuoDB license key is not effective: %s", output)
 	}
 }
+
+func GetDomainServersE(t *testing.T, namespace string, adminPod string) (map[string]NuoDBServer, error) {
+	options := k8s.NewKubectlOptions("", "", namespace)
+	output, err := k8s.RunKubectlAndGetOutputE(t, options, "exec", adminPod, "--",
+		"nuocmd", "--show-json", "get", "servers")
+	if err == nil {
+		err, servers := UnmarshalDomainServers(output)
+		return servers, err
+	}
+	return nil, err
+}
+
+func GetDomainLeaderE(t *testing.T, namespace string, adminPod string) (string, error) {
+	servers, err := GetDomainServersE(t, namespace, adminPod)
+	if err != nil {
+		return "", err
+	}
+	for _, server := range servers {
+		if server.ConnectedState.State == "Connected" && server.LocalRoleInfo.LeaderServerId != "" {
+			return server.LocalRoleInfo.LeaderServerId, nil
+		}
+	}
+	return "", errors.New("no leader found")
+}
+
+func AwaitDomainLeader(t *testing.T, namespace string, adminPod string, actionFn func(string),
```

Comment:
its a bit weird that you are calling the lambda inside of the Await function but it has no impact on its logic. It does not define when the await succeeds, it just executes the lambda exactly once before returning. Do you have to pass the lambda down, or can you just call it in the caller?

Reply:
```func AwaitDomainLeader(t *testing.T, namespace string, adminPod string, timeout time.Duration) (leader string){
	Await(t, func() bool {
		var err error
		leader, err = GetDomainLeaderE(t, namespace, adminPod)
		if err != nil {
			return false
		}
		
		return true
	}, timeout)
	
	return
}```

Reply:
The `testlib.Await` function doesn't support returning a value, hence we need to pass another lambda to be executed once a leader is found. The `actionFn` can be any callback including an empty one. If you prefer I can rename the routine to something like `ExecuteWithDomainLeader`.

Reply:
`Await` doesnt have to return a value. You can bind to a local variable.

Reply:
I looked again at the example you provided and got your point. Thanks!